### PR TITLE
Fix `docker compose` configuration with the new tracker container image

### PR DESCRIPTION
**Pull request by josecelano**
The [Torrust Tracker container image has changed](https://github.com/torrust/torrust-tracker/commit/7d133aef619bd84c56b216caddf2e766aa2a739c). 

We have to upgrade the configuration to use the latest version.

Integration tests fail for pull requests because the tracker container cannot start.

The significant changes introduced were:

- Rename some env vars: TORRUST_TRACKER_API_TOKEN->TORRUST_TRACKER_API_ADMIN_TOKEN
- Database folder was moved to a new location

### CODE REVIEW

**Comment by josecelano** [comment]
ACK d0e3d7320637754b365d700aaf060bf4a8cc2b41

**Comment by josecelano** [comment]
You can see here on example of workflow failing:

https://github.com/torrust/torrust-index-backend/actions/runs/6100451779/job/16556658907?pr=269

You can see in the `docker ps` output that the tracker service is not running:

```
Container torrust-mysql-1 is healthy
CONTAINER ID   IMAGE                       COMMAND                  CREATED          STATUS                    PORTS                                                                                  NAMES
8adcaa70a0ab   torrust-idx-back            "cargo run"              34 seconds ago   Up 32 seconds             0.0.0.0:3001->3001/tcp, :::3001->3001/tcp                                              torrust-idx-back-1
fa41beb4de15   mysql:8.0                   "docker-entrypoint.s…"   34 seconds ago   Up 33 seconds (healthy)   0.0.0.0:3306->3306/tcp, :::3306->3306/tcp, 33060/tcp                                   torrust-mysql-1
4842ffb5e42a   dockage/mailcatcher:0.8.2   "entrypoint mailcatc…"   34 seconds ago   Up 33 seconds             0.0.0.0:1025->1025/tcp, :::1025->1025/tcp, 0.0.0.0:1080->1080/tcp, :::1080->1080/tcp   torrust-mailcatcher-1
```

File: docker/bin/e2e/mysql/e2e-env-reset.sh [hunk 1]
```diff
@@ -19,15 +19,15 @@ mysql -h $MYSQL_HOST -u $MYSQL_USER -p$MYSQL_PASSWORD -e "DROP DATABASE IF EXIST
...
-mkdir -p "./storage/database"
+mkdir -p "./storage/tracker/lib/database"
 
 # Generate the sqlite database for the tracker if it does not exist
-if ! [ -f "./storage/database/torrust_tracker_e2e_testing.db" ]; then
-    touch ./storage/database/torrust_tracker_e2e_testing.db
-    echo ";" | sqlite3 ./storage/database/torrust_tracker_e2e_testing.db
+if ! [ -f "./storage/tracker/lib/database/torrust_tracker_e2e_testing.db" ]; then
+    touch ./storage/tracker/lib/database/torrust_tracker_e2e_testing.db
+    echo ";" | sqlite3 ./storage/tracker/lib/database/torrust_tracker_e2e_testing.db
```
Comment (da2ce7) [inline docker/bin/e2e/mysql/e2e-env-reset.sh:30]: I recommend that you create the empty database with this command:

https://github.com/torrust/torrust-tracker/blob/698890a0b1dcf696013af640a131a7c146d295ed/Containerfile#L20-L21

File: docker/bin/e2e/sqlite/e2e-env-reset.sh [hunk 2]
```diff
@@ -18,9 +18,9 @@ if ! [ -f "./storage/database/torrust_index_backend_e2e_testing.db" ]; then
 fi
 
 # Generate the sqlite database for the tracker if it does not exist
-if ! [ -f "./storage/database/torrust_tracker_e2e_testing.db" ]; then
-    touch ./storage/database/torrust_tracker_e2e_testing.db
-    echo ";" | sqlite3 ./storage/database/torrust_tracker_e2e_testing.db
+if ! [ -f "./storage/tracker/lib/database/torrust_tracker_e2e_testing.db" ]; then
+    touch ./storage/tracker/lib/database/torrust_tracker_e2e_testing.db
+    echo ";" | sqlite3 ./storage/tracker/lib/database/torrust_tracker_e2e_testing.db
```
Comment (da2ce7) [inline docker/bin/e2e/sqlite/e2e-env-reset.sh:23]: again here.

**Review by da2ce7 — APPROVED** [review]

File: docker/bin/e2e/mysql/e2e-env-reset.sh (hunk 1 above)
Comment (josecelano) [reply docker/bin/e2e/mysql/e2e-env-reset.sh:30]: Yes, thank you! I saw the other day you were using it. I will change it.

File: docker/bin/e2e/sqlite/e2e-env-reset.sh [hunk 3]
```diff
@@ -5,22 +5,20 @@
...
 rm -f ./storage/database/torrust_index_backend_e2e_testing.db
-rm -f ./storage/database/torrust_tracker_e2e_testing.db
+rm -f ./storage/tracker/lib/database/torrust_tracker_e2e_testing.db
 
 # Generate storage directory if it does not exist
 mkdir -p "./storage/database"
 
 # Generate the sqlite database for the index backend if it does not exist
 if ! [ -f "./storage/database/torrust_index_backend_e2e_testing.db" ]; then
     # todo: it should get the path from config.toml and only do it when we use sqlite
```
Comment (da2ce7) [inline docker/bin/e2e/sqlite/e2e-env-reset.sh:15]: This todo can be removed.

**Review by da2ce7 — APPROVED** [review]
Looks Good.

File: docker/bin/e2e/sqlite/e2e-env-reset.sh (hunk 3 above)
Comment (josecelano) [reply docker/bin/e2e/sqlite/e2e-env-reset.sh:15]: Hi @da2ce7  I meant that there should be only one place to define the database. I think the `config-idx-back.sqlite.local.toml` or the `config-idx-back.sqlite.local.toml` files` is the original place where that info is located. At least now we have another env var
`TORRUST_IDX_BACK_CONFIG` which is better than the hardcoded value. Anyway, I think having to places to define the same value can lead to confusion:

- TORRUST_IDX_BACK_CONFIG
- TORRUST_IDX_BACK_MYSQL_DATABASE

Having the second value overwriting the first one is something that the user might expect, but I think we have to find an explicit convention for this behaviour. It's something to consider in the [configuration overhaul](https://github.com/torrust/torrust-tracker/issues/401).

For now, I will remove the todo and I will use the env var TORRUST_IDX_BACK_MYSQL_DATABASE in all the scripts.

**Comment by josecelano** [comment]
ACk bf95d854ec5f360de310a53d69f9c5748914a53

**Pull request merged**